lib/tx_build/preconditions.ex

```diff
@@ -0,0 +1,185 @@
+defmodule Stellar.TxBuild.Preconditions do
+  @moduledoc """
+  `Preconditions` struct definition.
+  """
+  alias Stellar.TxBuild.{
+    TimeBounds,
+    LedgerBounds,
+    SequenceNumber,
+    SignerKey
+  }
+
+  alias StellarBase.XDR.TimeBounds, as: TimeBoundsXDR
+
+  alias StellarBase.XDR.{
+    Preconditions,
+    PreconditionsV2,
+    PreconditionType,
+    Duration,
+    OptionalSequenceNumber,
+    SignerKeyList,
+    TimePoint,
+    UInt32,
+    Void
+  }
+
+  @behaviour Stellar.TxBuild.XDR
+
+  @type precond_v2 :: %{
```

Comment:
@CristhianRodriguezMolina @FelipeGuzmanSierra Let's use a keyword list for this specification.